src/main.ts

```diff
@@ -553,6 +557,71 @@ class SettingTab extends PluginSettingTab {
           });
         });
   }
+
+  addCustomCommandsSetting(): void {
+    this.containerEl.createEl('h2', {text: 'Custom Commands Settings'});
+    function arrayMove(arr:string[], fromIndex:number, toIndex:number):void {
+      if (toIndex < 0 || toIndex === arr.length) {
+        return;
+      }
+      const element = arr[fromIndex];
+      arr[fromIndex] = arr[toIndex];
+      arr[toIndex] = element;
+    }
+    new Setting(this.containerEl)
+        .addButton((cb)=>{
+          cb.setButtonText('Add new command')
+              .setCta()
+              .onClick(()=>{
+                this.plugin.settings.lintCommands.push('');
+                this.plugin.saveSettings();
+                this.display();
+              });
+        });
+
+    this.plugin.settings.lintCommands.forEach((command, index) => {
+      new Setting(this.containerEl)
+          .addSearch((cb) => {
+            new CommandSuggester(this.app, cb.inputEl);
+            cb.setPlaceholder('Example: folder1/template_file')
+                .setValue(command)
+                .onChange((newCommand) => {
+                  if (newCommand && this.plugin.settings.lintCommands.contains(newCommand)) {
+                    logError('This command is already exist!', new Error('This command is already exist!'));
```

Comment:
The following is a minor grammar fix and a change in wording in the log to make it clearer and not repeat the same info twice:
```suggestion
                    logError('You cannot add the same command to the list of custom rules twice.', new Error('this command already exists!'));
```